### DISCUSSION TITLE
New Lambda Function URL capability; router functionality?

### DISCUSSION THREAD

**Discussion by rkulagowski** [aws-powertools/powertools-lambda-python#1097] [Q&A] [answered]
https://aws.amazon.com/blogs/aws/announcing-aws-lambda-function-urls-built-in-https-endpoints-for-single-function-microservices/

If I'd like to migrate from API Gateway to direct invocation, can I use one of the router functions (decorators) to send various HTTP methods to the appropriate function in my monolith? I don't want to add Flask if I don't have to.

**Comment by michaelbrewer** [comment]
@rkulagowski - this already works via the [APIGatewayHttpResolver](https://awslabs.github.io/aws-lambda-powertools-python/latest/core/event_handler/api_gateway/#api-gateway-http-api
)

**Reply by michaelbrewer** [reply]
Otherwise there are 2 open issues to support this officially for data classes and the event handler:

- https://github.com/awslabs/aws-lambda-powertools-python/issues/1141
- https://github.com/awslabs/aws-lambda-powertools-python/issues/1142

**Reply by deric4** [reply]
ah nice! after reading your comments in #1141 and #1142 I realize where I incorrectly assumed this wasn't supported when playing around trying to implement a [custom cloudfront origin using a lambda function url](https://aws.amazon.com/blogs/networking-and-content-delivery/using-amazon-cloudfront-with-aws-lambda-as-origin-to-accelerate-your-web-applications/).

I hit this part of the docs first, and incorrectly concluded it wasn't supported
> You must have an existing [API Gateway Proxy integration](https://docs.aws.amazon.com/apigateway/latest/developerguide/set-up-lambda-proxy-integrations.html) or [ALB](https://docs.aws.amazon.com/elasticloadbalancing/latest/application/lambda-functions.html) configured to invoke your Lambda function. There is no additional permissions or dependencies required to use this utility.

https://awslabs.github.io/aws-lambda-powertools-python/latest/core/event_handler/api_gateway/#required-resources

I'm not sure how accurate my current mental model  of there being an  _implicit_ API Gateway proxying traffic to my lambda is, but running all of my logged events through `event_parser` with `APIGatewayProxyEventV2Model`  works  🥳

https://github.com/awslabs/aws-lambda-powertools-python/blob/c1955b87346010556645b59c0e934a12ed290beb/aws_lambda_powertools/utilities/parser/models/apigwv2.py#L59-L71

Thanks @michaelbrewer !

**Comment by heitorlessa** [comment]
We'll look into adding official support for Function URL as a new Resolver class after our pause - here's the feature request that we'd appreciate you giving a +1: https://github.com/awslabs/aws-lambda-powertools-python/issues/1142

In the meantime, [Michael's answer](https://github.com/awslabs/aws-lambda-powertools-python/discussions/1097#discussioncomment-2541169) should work. Function URL uses HTTP API payload v2 with minor/non-breaking changes.

Thank you everyone

**Comment by heitorlessa** [comment, marked as answer]
Closing as this was launched back then: https://docs.powertools.aws.dev/lambda/python/latest/core/event_handler/api_gateway/#lambda-function-url
